software/contrib/largefont_demo.py

```diff
@@ -0,0 +1,58 @@
+"""
+Large font demo
+author: François Georgy (https://github.com/francoisgeorgy)
+date: 2023-02-15
+
+Use this script to test and demo the support for large fonts.
+
+See the largefont_writer module for information about how to import your own fonts.
+"""
+import time
+
+from europi_script import EuroPiScript
+from europi import oled, OLED_WIDTH, OLED_HEIGHT
+import freesans14
+import freesans17
+import freesans20
+import freesans24
+
+
+class LargeFontDemo(EuroPiScript):
+    @classmethod
+    def display_name(cls):
+        return "Large font demo"
+
+    def __init__(self):
+        super().__init__()
+        self.demo = 4
+        self.boxed = True  # alternate one round boxed, one round unboxed.
+
+    def update_demo(self):
+        self.demo = (self.demo + 1) % 5
+        if self.demo == 0:
+            oled.centre_text("Default\nmonospaced\n8x8 font")
+            self.boxed = not self.boxed
+        elif self.demo == 1:
+            oled.centre_text("FreeSans 14\n1234567890123456", font=freesans14)
+        elif self.demo == 2:
+            oled.centre_text("FreeSans 17", font=freesans17)
+        elif self.demo == 3:
+            oled.centre_text("FreeSans 20", font=freesans20)
+        elif self.demo == 4:
+            oled.centre_text("FreeS. 24", font=freesans24)
```

Comment:
[discussion] The font at size 20 and 24 seem too large to be useful on this size OLED. 24 in particular looks like it'd only fit 9 or 10 characters. I'd like to avoid adding fonts that never get used.

14 seems great to me as it's the perfect size for two line of text, which complements the existing font's three lines.

I'd suggest that we limit this PR to adding just one or two new fonts, for now. And adding more when scripts show some need for them. But I'd like to hear other opinions on this.

Reply:
Well, I both agree and disagree with you. Size 24 is really big but I use it in one of my project because I want to have a value easy visible and readable, even from a short distance.

Like I said in my previous comment, I think it's better if the font themselves remain outside the firmware. And I think you agree with me.

Reply:
I'm inclined to agree with @francoisgeorgy that having the options would be great, however we need to be careful about how much memory we're using. For context, the pico only has 256k RAM and 2mb flash (we only have about 1mb of flash available after installing MicroPython, and our firmware and scripts take up about 350k last I checked https://discord.com/channels/931297838804127794/931491803666993202/1037365430098075809). I think it's fine to leave them in now, but be mindful of how much memory we are using by loading all scripts, fonts, etc.

Reply:
> because I want to have a value easy visible and readable

I was thinking about text specifically, but I could see where this would be great for say a 2-3 digit number.

As for the memory / storage discussion: We're not loading fonts into RAM until they are used, so that should be fine. For the storage piece, I think that's one of the benefits of the experimental dir here. We can see what gets used and what doesn't and modify the initial set of sizes accordingly.